sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suite.scala

```diff
@@ -121,7 +152,16 @@ class HiveThriftServer2Suite extends FunSuite with Logging {
   }
 
   def startThriftServer(
+      thriftPort: Int,
+      serverStartTimeout: FiniteDuration = 1.minute,
+      httpMode: Boolean = false)(
+      f: => Unit): Unit = {
```

Comment:
Also, change `f: => Unit` to `f: (Int, Int) => Unit` and pass in server port and UI port here.